plugins/grype/surfactantplugin_grype.py

```diff
@@ -4,34 +4,55 @@
 # SPDX-License-Identifier: MIT
 import gzip
 import subprocess
+
+# Configure loguru to show logs at DEBUG level or higher
+import sys
 import tempfile
 from typing import List, Optional
 
 from loguru import logger
 
 import surfactant.plugin
+from surfactant.plugin.manager import (  # Import the helper function
+    get_plugin_manager,
+    is_plugin_blocked,
+)
 from surfactant.sbomtypes import SBOM, Software
 
+logger.remove()  # Remove the default logger configuration
+logger.add(sys.stdout, level="DEBUG")  # Add a new logger that outputs to console
+
 
 def check_if_grype_installed() -> bool:
     try:
         result = subprocess.run(["grype", "--help"], capture_output=True, check=False).returncode
+        if result == 0:
+            logger.info("Grype is installed and ready to use.")
+        else:
+            logger.warning("Grype is not installed or not functioning correctly.")
+        return result == 0
     except FileNotFoundError:
-        result = 1
-    if result != 0:
-        logger.warning("Install grype for the grype plugin to run")
-    return result == 0
+        logger.error("Grype is not installed. Please install it to use this plugin.")
+        return False
 
 
-disable_plugin = not check_if_grype_installed()
+# Initialize the plugin manager
+plugin_manager = get_plugin_manager()
+disable_plugin = not check_if_grype_installed() or is_plugin_blocked(plugin_manager, __name__)
```

Comment:
```suggestion
disable_plugin = not check_if_grype_installed() or is_plugin_blocked(plugin_manager, __name__)
```

If the plugin is blocked the extract_file_info hook should never get called, so  the value of`disable_plugin` will never be checked in the case that it is set to True due to the is_plugin_blocked call.